src/definition.rs

```diff
@@ -134,6 +143,38 @@ impl TemplateDefinition {
     }
 }
 
+fn replace_with_previous_responses(variables: HashSet<&str>, previous_responses: &HashMap<String, Value>, to_replace: &String) -> String {
+    let mut cloned_str = to_replace.clone();
+
+    for variable in variables.iter() {
+        let var_without_braces = variable.trim_start_matches("{{").trim_end_matches("}}");
+        if let Some(value) = previous_responses.get(var_without_braces) {
+            cloned_str = match value {
+                Value::String(v) => str::replace(&cloned_str, variable, &v),
+                Value::Integer(v) => str::replace(&cloned_str, variable, &v.to_string()),
+                Value::Float(v) => str::replace(&cloned_str, variable, &v.to_string()),
+                Value::Boolean(v) => str::replace(&cloned_str, variable, &v.to_string()),
+                _ => cloned_str
+            }
+        }
+    }
+
+    cloned_str
+}
+
+fn has_template_variables<'a>(s: &'a String, _already_vars: &HashMap<String, Value>) -> Option<HashSet<&'a str>> {
+    let re = Regex::new(r"\{\{(?:[a-zA-Z][0-9a-zA-Z_]*)\}\}").unwrap();
```

Comment:
that's going to compile that regex every time this function is called, you can lift it in some lazy static or once_cell, whatever this crate is using (or add one of these libs to it)